src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Reissuance/ReissuanceConsent/HealthCertificateReissuanceConsentViewModel.swift

```diff
@@ -227,6 +264,24 @@ final class HealthCertificateReissuanceConsentViewModel {
 		}
 	}
 	
+	private func filterAccompanyingCertificates(
+		certificates: [DCCCertificateContainerExtended],
+		certifiedPerson: HealthCertifiedPerson
+	) -> [HealthCertificate] {
+		var finalArray = [DCCCertificateContainer]()
+		let reissuanceCertificates = certificates.compactMap({ $0.certificateToReissue })
+		for certificate in certificates {
+			for accompanyingCertificate in certificate.accompanyingCertificates {
+				if !reissuanceCertificates.contains(accompanyingCertificate) && !finalArray.contains(accompanyingCertificate) {
+					finalArray.append(accompanyingCertificate)
+				}
+			}
+		}
+		return finalArray.compactMap({
+			certifiedPerson.healthCertificate(for: $0.certificateRef)
+		}).sorted(by: >)
```

Comment:
Can we have a unit test for this?

Reply:
I also tried to write down a functional alternative to see if it's easer to read/understand, not sure if it is 😄 
```suggestion
		certificates
			.flatMap {
				$0.accompanyingCertificates
			}
			.filter {
				!certificates
					.map { $0.certificateToReissue }
					.contains($0)
			}
			.uniqued()
			.compactMap {
				certifiedPerson.healthCertificate(for: $0.certificateRef)
			}
			.sorted(by: >)
```
Also not sure if it's really equivalent without a unit test 😉

Reply:
We have a separate ticket for this unit test but i guess i can just do it in this PR